frontend/simple/js/state.js

```diff
@@ -5,22 +5,123 @@
+  updateCurrentGroupLog (state, log) {
+    if (Array.isArray(log.offset)) {
+      state.currentGroupLog = log.log
+      state.offset = log.offset
+    } else {
+      state.currentGroupLog = log
+    }
+  },
+  updateAvailableGroups (state, available) {
+    state.availableGroups = available
+  },
+  updateSocket (state, socket) {
+    state.socket = socket
+  }
+}
+
+export const actions = {
+  async createGroup ({commit, state}, group) {
+    let log = await EventLog.createEventLogFromGroup(group)
+    let available = state.availableGroups
+    if (state.socket) {
+      await joinRoom(state.socket, group)
+    }
+    commit('updateCurrentGroupLog', log)
+    available.push(log.currentLogPosition)
+    commit('updateAvailableGroups', available)
+  },
+  // TODO: Connect this to user credentials
+  // limits to available groups for user
+  async loadGroup ({commit, state}, group) {
+    let available = state.availableGroups
+    if (available.find((grp) => { return group === grp })) {
+      if (state.socket) {
+        await joinRoom(state.socket, group)
+      }
+      let log = await EventLog.getEventLogForGroupId(group)
+      let offset = []
+      commit('updateCurrentGroupLog', {log, offset})
+    }
+  },
+  async appendLog ({commit, state}, event) {
+    let log = await EventLog.addItemToLog(state.currentGroupLog, event)
+    commit('updateCurrentGroupLog', log)
+  },
+  async receiveEvent ({commit, state}, msg) {
+    // check if this is a log update
+    if (msg.data && msg.data.entry && msg.data.entry.parentHash) {
+      let available = state.availableGroups
+      // check to see if this event is for an available group
+      if (available.find((grp) => { return msg.data.groupId === grp })) {
+        let log = await EventLog.getEventLogForGroupId(msg.data.groupId)
+        log = await EventLog.addItemToLog(log, msg.data.entry.data, true)
+        // update curren.t if the event is for the current group
+        if (state.currentGroupLog && state.currentGroupLog.currentLogPosition === msg.data.groupId) {
+          commit('updateCurrentGroupLog', log)
+        }
+      }
+    }
+  },
+  async backward ({commit, state}) {
+    if (state.currentGroupLog) {
+      let entry = await EventLog.getItemFromLog(state.currentGroupLog, state.currentGroupLog.currentLogPosition)
+      let log = {groupId: state.currentGroupLog.groupId, currentLogPosition: entry.parentHash}
+      let offset = state.offset.slice(0)
+      offset.push(state.currentGroupLog.currentLogPosition)
+      commit('updateCurrentGroupLog', {log, offset})
+    }
+  },
+  async forward ({commit, state}) {
+    if (state.currentGroupLog && state.offset.length) {
+      let offset = state.offset.slice(0)
+      let prior = offset.pop()
+      let log = {groupId: state.currentGroupLog.groupId, currentLogPosition: prior}
+      commit('updateCurrentGroupLog', {log, offset})
+    }
+  },
+  loggedIn: state => !!state.loggedInUser,
```

Comment:
This must be a relic from before. I don't think this is how actions work now. So this needs to be rewritten or removed.